src/app/clusters/scenes-server/SceneTableImpl.cpp

```diff
@@ -737,11 +737,17 @@ CHIP_ERROR DefaultSceneTableImpl::SceneSaveEFS(SceneTableEntry & scene)
 {
     if (!HandlerListEmpty())
     {
-        uint8_t clusterCount = 0;
-        clusterId cArray[kMaxClustersPerScene];
-        Span<clusterId> cSpan(cArray);
-        clusterCount = GetClustersFromEndpoint(cArray, kMaxClustersPerScene);
-        cSpan.reduce_size(clusterCount);
+        uint8_t clusterCount = GetClusterCountFromEndpoint();
+        clusterId * cBuffer  = static_cast<clusterId *>(chip::Platform::MemoryCalloc(sizeof(clusterId), clusterCount));
+        VerifyOrReturnError(nullptr != cBuffer, CHIP_ERROR_NO_MEMORY);
+
+        clusterCount = GetClustersFromEndpoint(cBuffer, clusterCount);
+        chip::Platform::MemoryRealloc(cBuffer, clusterCount);
+
+        // If there were no supported clusters on the endpoint, returns no error
+        VerifyOrReturnError(nullptr != cBuffer, CHIP_NO_ERROR);
```

Comment:
I don't think realloc guarantees either "only returns null on 0 size" or "always returns null on 0 size", so this comment's relationship to this code is a bit tenuous.